tools/gulp/tasks/publish/branch-check.ts

```diff
@@ -1,41 +1,34 @@
-import {bold} from 'chalk';
 import {spawnSync} from 'child_process';
 import {buildConfig} from 'material2-build-tools';
 
 /** Regular expression that matches version names and the individual version segments. */
 export const versionNameRegex = /^(\d+)\.(\d+)\.(\d+)(?:-(alpha|beta|rc)\.(\d)+)?/;
 
-/** Regular expression that matches publish branch names and their Semver digits. */
-const publishBranchNameRegex = /^([0-9]+)\.([x0-9]+)(?:\.([x0-9]+))?$/;
-
 /** Checks if the specified version can be released from the current Git branch. */
 export function checkPublishBranch(version: string) {
   const versionType = getSemverVersionType(version);
   const branchName = spawnSync('git', ['symbolic-ref', '--short', 'HEAD'],
     {cwd: buildConfig.projectDir}).stdout.toString().trim();
 
-  if (branchName === 'master') {
-    if (versionType === 'major') {
-      return;
-    }
-
-    throw `Publishing of "${versionType}" releases should not happen inside of the ` +
-        `${bold('master')} branch.`;
-  }
 
-  const branchNameMatch = branchName.match(publishBranchNameRegex) || [];
-  const branchDigits = branchNameMatch.slice(1, 4);
+  // TODO(devversion): also check the the local branch's HEAD sha matches upstream.
+  // TODO(devversion): also check that the version is a single increment of the previous release.
+  // TODO(devversion): minor releases can also be published from master if there isn't a minor
+  // branch for that major range yet.
+  const [major, minor] = version.split('.');
 
-  if (branchDigits[2] === 'x' && versionType !== 'patch') {
-    throw `Cannot publish a "${versionType}" release inside of a patch branch (${branchName})`;
+  let expectedBranch = '';
+  if (versionType === 'major') {
+    expectedBranch = 'master';
+  } else if (versionType === 'minor') {
+    expectedBranch = `${major}.x.x`;
```

Comment:
Just a question: right now isn't the branch always `6.x`? instead of `6.x.x`?

Reply:
Ah, good catch.